core/src/test/java/org/elasticsearch/indices/IndicesServiceTests.java

```diff
@@ -81,6 +81,8 @@ public void testCanDeleteShardContent() {
         assertFalse("shard is allocated", indicesService.canDeleteShardContent(shardId, test.getIndexSettings()));
         test.removeShard(0, "boom");
         assertTrue("shard is removed", indicesService.canDeleteShardContent(shardId, test.getIndexSettings()));
+        ShardId notAllocated = new ShardId(test.index(), 100);
+        assertFalse("shard doesn't nor hasn't on this node", indicesService.canDeleteShardContent(notAllocated, test.getIndexSettings()));
```

Comment:
the amount of negatives here is impressive :) - "shard that was never on this node should be deletable " ?